src/util/cilfacade.ml

```diff
@@ -80,71 +80,45 @@ let end_basic_blocks f =
   let thisVisitor = new allBBVisitor in
   visitCilFileSameGlobals thisVisitor f
 
+class removeLabelsVisitor = object
+  inherit nopCilVisitor
+
+  val nestingDepth = ref 0
+
+  method! vstmt s =
+    let new_labels = List.filter_map (function Label(str,loc,b) -> None | x -> Some x) s.labels in
+    match s.skind with
+    | Loop _ -> nestingDepth := !nestingDepth+1; ChangeDoChildrenPost({s with labels = new_labels}, fun x -> nestingDepth := !nestingDepth-1; x)
+    | Continue loc -> if !nestingDepth = 0 then
+        ChangeTo({s with labels = new_labels; skind = Break loc})
```

Comment:
What's this about? It's not just removing labels.

Reply:
All unrollings are now in a `while(true)` loop. Top-level `Continue`s inside this loop now should continue to the next (unrolled) iteration, i.e. break out of the infinite loop.

Reply:
Won't the top level breaks then be wrong? They would have to jump over all the loops.

Reply:
Yes, you are right of course. I had thought about needing it and the somehow forgot. I guess this is part of the reason, why it still so simple - simply wrong :wink: